### PR TITLE
Create bundle.md

### PR DESCRIPTION
**Pull request by loosebazooka**
A human readable bundle specification, specifically for the public good instances

I'm not sure doing only public good is idea, but I think it removes some of the other more confusing things from the spec.

The included github attestation example is from conformance -- maybe there's a better real world example somewhere?

### CODE REVIEW

File: content/en/about/bundle.md [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+[sigstore/protobuf-specs](https://github.com/sigstore/protobuf-specs).
+
+## Bundle
+
+A Sigstore bundle is everything required to verify a signature on an artifact. This
+is satisfied by the **Verification Material** and signature **Content**.
+
+### Verification Material
+
+This is key materical used to verify signatures along with supporting metadata like transparency log entries and timestamps. Bundles must include at least one transpareny log's signed entry timestamp or an rfc3161 timestamp to provide proof of signing time.
```
Comment (haydentherapper) [inline content/en/about/bundle.md:21]:
```suggestion
This is key material used to verify signatures along with supporting metadata like transparency log entries and timestamps. Bundles must include at least one transparency log's signed entry timestamp or an [RFC3161](https://www.ietf.org/rfc/rfc3161.txt) timestamp to provide proof of signing time.
```

File: content/en/about/bundle.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+#### Key Material
+
+A single X.509 leaf certificate conveying the signing key and containing [extensions](https://github.com/sigstore/fulcio/blob/main/docs/oid-info.md) 
+for identities consumed at verification time. This is the recommended `"verificationMaterial"` type
+for use with the public Sigstore infrastructure.
```
Comment (haydentherapper) [inline content/en/about/bundle.md:27]: Can we mention `PublicKeyIdentifier` as another option? Using a self-managed key and recording the signature in the public Rekor instance is a supported workflow.

File: content/en/about/bundle.md [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+  "certificate": {
+    "rawBytes": "<Base64(DER bytes)>"
+  }
+}
+```
+
+#### Transparency Log Entries
+
+One or more transparency logs entries to provide proof of inclusion in a public log and optionally a timestamp to
+validate signing occurred at a valid point in time.
```
Comment (haydentherapper) [inline content/en/about/bundle.md:40]: I'd define what we mean by "valid point in time", as in a user was in possession of a signing key and used it to produce a signature, and that signing key was certified by a CA, and this is recorded while the certificate is valid in the log.

File: content/en/about/bundle.md [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+      },
+      "canonicalizedBody": "<Base64(JSON)>"
+    },
+  ]
+}
+```
+
+#### Timestamp
+
+Zero or more [RFC3161](https://www.ietf.org/rfc/rfc3161.txt) timestamps to validate signing occurred at a valid point in time
```
Comment (haydentherapper) [inline content/en/about/bundle.md:79]: Same here to clarify what "valid" means

File: content/en/about/bundle.md [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+    ]
+  }
+}
+```
+
+
+### Content
+
+This is the signature data for which the Verification Material is defined over. It must be one of
+**Message Signature** over an artifact hash or a **DSSE** envelope for for attestations.
```
Comment (haydentherapper) [inline content/en/about/bundle.md:97]:
```suggestion
**Message Signature** over an artifact hash or a **DSSE** envelope for attestations.
```

File: content/en/about/bundle.md [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+    "algorithm": "<HASH ALGORITHM>",
+    "digest": "<Base64(DIGEST)>"
+  },
+  "signature": "<Base64(SIGNATURE)>"
+}
+```
+
+#### DSSE
+ 
+A DSSE envelope can contain arbitrary payloads. Currently clients only process the
```
Comment (haydentherapper) [inline content/en/about/bundle.md:117]:
```suggestion
A DSSE envelope can contain arbitrary payloads. Currently Sigstore clients only process the
```

**Review by haydentherapper — COMMENTED** [review]
This is great!

Do you think we should discuss what to do with a bundle at all, touching on the verification aspect, e.g. needing to verify identities, meet thresholds?

File: content/en/about/bundle.md (hunk 2 above)
Comment (loosebazooka) [reply content/en/about/bundle.md:29]: done

File: content/en/about/bundle.md (hunk 4 above)
Comment (loosebazooka) [reply content/en/about/bundle.md]: done

File: content/en/about/bundle.md (hunk 3 above)
Comment (loosebazooka) [reply content/en/about/bundle.md]: done

**Review by loosebazooka — COMMENTED** [review]
moved signing time to "Verification Material" section.

**Review by haydentherapper — APPROVED** [review]
Thanks, this is great!

**Review by haydentherapper — APPROVED** [review]

**Review by kommendorkapten — APPROVED** [review]
Awesome!

**Comment by haydentherapper** [comment]
Will let @steiza  review, then I’ll merge

File: content/en/about/bundle.md [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+"verificationMaterial": {
+  "publicKeyIdentifier": {
+    "hint": "<HINT>"
+  }
+}
+```
+
+#### Transparency Log Entries
+
+One or more transparency logs entries to provide proof of inclusion in a public log and optionally a timestamp to
```
Comment (steiza) [inline content/en/about/bundle.md:54]: Minor: a transparency log entry isn't required for a bundle, should we say "Zero or more..." like we do below for Timestamps?

We already say above:
> bundles must include at least one transparency log's signed entry timestamp or an RFC3161 timestamp

**Review by steiza — APPROVED** [review]
One minor non-blocking comment

File: content/en/about/bundle.md (hunk 7 above)
Comment (loosebazooka) [reply content/en/about/bundle.md:54]: Oh I always thought a log entry was required but you don't have to trust the logs clock (so you can ignore the SET and use the rfc3161 timestamp). Maybe that's only true for the public usecase?

Comment (haydentherapper) [reply content/en/about/bundle.md:54]: Log entries are not required - https://github.com/sigstore/protobuf-specs/blob/main/protos/sigstore_bundle.proto#L105

I think we should word it that it is expected that bundles contain log entries for public consumption.

Comment (loosebazooka) [reply content/en/about/bundle.md]: updated this section a bit

**Review by haydentherapper — APPROVED** [review]

**Pull request merged**